cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -440,6 +466,16 @@ func create(ctx context.Context, flags *pflag.FlagSet) error {
 		return err
 	}
 
+	apiPort, err := getDynamicTCPPort()
```

Comment:
we need to allocate only if the provisioner is docker...

Reply:
in fact, we could probably hide it more under the `provisioner` interface, and make only Docker one do that